### PR TITLE
Add missing reflection doc

### PR DESCRIPTION
**Pull request by steveharter**
Add missing doc based on https://github.com/dotnet/runtime/issues/88571#issuecomment-1721513012

There is also reflection.emit changes in that issue, that will be addressed in a different PR.

### CODE REVIEW

File: xml/System.Reflection/DispatchProxy.xml
Comment (carlossanlop) [inline xml/System.Reflection/DispatchProxy.xml:133]:
```suggestion
          <paramref name="interfaceType" /> or <paramref name="proxyType" /> is <see langword="null"/>.</exception>
```

File: xml/System.Reflection/TypeDelegator.xml
Comment (carlossanlop) [inline xml/System.Reflection/TypeDelegator.xml:1060]:
```suggestion
        <returns>
          <para>An array of <see cref="T:System.Type" /> objects representing all the calling conventions for the current function pointer <see cref="T:System.Type" />.</para>
          <para>-or-</para>
          <para>An empty array of type <see cref="T:System.Type" />, if no calling conventions are defined for the current function pointer <see cref="T:System.Type" />.</para>
          <para>-or-</para>
          <para>An empty array of type <see cref="T:System.Type" />, if the current function pointer <see cref="T:System.Type" /> is not a modified <see cref="T:System.Type" />. A modified <see cref="T:System.Type" /> is obtained from <see cref="M:System.Reflection.FieldInfo.GetModifiedFieldType" />, <see cref="M:System.Reflection.PropertyInfo.GetModifiedPropertyType" />, or <see cref="M:System.Reflection.ParameterInfo.GetModifiedParameterType" />.</para>
        </returns>
```

File: xml/System.Reflection/TypeDelegator.xml
Comment (carlossanlop) [inline xml/System.Reflection/TypeDelegator.xml:1068]: Optional: We can use plain xml for remarks. It's easier to handle here and in triple slash. We want to gradually move to using that instead of markdown.

```suggestion
        <remarks>This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <see cref="N:System.Runtime.CompilerServices"/> namespace.</remarks>
```

File: xml/System.Reflection/TypeDelegator.xml
Comment (carlossanlop) [inline xml/System.Reflection/TypeDelegator.xml:1100]: The property is incomplete. Did you mean P:System.Type.IsFunctionPointer like in the exception below?

File: xml/System.Reflection/DispatchProxy.xml
Comment (gewarren) [inline xml/System.Reflection/DispatchProxy.xml:135]:
```suggestion
          <paramref name="interfaceType" /> is a class

-or-

<paramref name="proxyType" /> is sealed or abstract, or doesn't inherit from the <see cref="System.Reflection.DispatchProxy" /> type or have a parameterless constructor.</exception>
```

File: xml/System.Reflection/TypeDelegator.xml
Comment (carlossanlop) [inline xml/System.Reflection/TypeDelegator.xml:1098]:
```suggestion
        <returns>
          <para>An array of <see cref="T:System.Type" /> objects representing all the parameter types for the current function pointer <see cref="T:System.Type" />.</para>
          <para>-or-</para>
          <para>An empty array of type <see cref="T:System.Type" />, if no parameters are defined for the current function pointer <see cref="T:System.Type" />.</para>
        </returns>
```

**Review by carlossanlop — APPROVED** [review]
Left some suggestions for you to consider. Otherwise, LGTM.

**Review by gewarren — APPROVED** [review]

File: xml/System.Reflection/TypeDelegator.xml
Comment (steveharter) [reply xml/System.Reflection/TypeDelegator.xml:1100]: Yes thanks

File: xml/System.Reflection/DispatchProxy.xml
Comment (carlossanlop) [inline xml/System.Reflection/DispatchProxy.xml:137]:
```suggestion
        <exception cref="T:System.ArgumentException">
          <para><paramref name="interfaceType" /> is a class.</para>
          <para>-or-</para>
          <para><paramref name="proxyType" /> is sealed or abstract, or doesn't inherit from the <see cref="T:System.Reflection.DispatchProxy" /> type or has a parameterless constructor.</para>
        </exception>
```

File: xml/System.Reflection/FieldInfo.xml
Comment (carlossanlop) [inline xml/System.Reflection/FieldInfo.xml:673]: I can't offer a full suggestion to replace the remarks with plain xml. For that reason, my suggestion for this case will be to move it back to xref, because it's throwing a warning in CI.

```suggestion
 This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace. Also see the metadata specification in Partition II of the [Common Language Infrastructure (CLI) documentation](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/).  
```

File: xml/System.Reflection/ParameterInfo.xml
Comment (carlossanlop) [inline xml/System.Reflection/ParameterInfo.xml:739]:
```suggestion
 This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace. Also see the metadata specification in Partition II of the [Common Language Infrastructure (CLI) documentation](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/).  
```

File: xml/System.Reflection/PropertyInfo.xml
Comment (carlossanlop) [inline xml/System.Reflection/PropertyInfo.xml:1002]:
```suggestion
 This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace. Also see the metadata specification in Partition II of the [Common Language Infrastructure (CLI) documentation](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/).  
```

File: xml/System.Reflection/TypeDelegator.xml
Comment (carlossanlop) [inline xml/System.Reflection/TypeDelegator.xml:1066]: To be able to use `see cref`, these remarks need to be moved from markdown to xml:

```suggestion
        <remarks>This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <see cref="N:System.Runtime.CompilerServices"/> namespace.</remarks>
```

File: xml/System/Type.xml
Comment (carlossanlop) [inline xml/System/Type.xml:7672]: Same. GitHub does not let me offer a full suggestion to convert the whole markdown section to xml, so we need to revert this:

```suggestion
 The <xref:System.Type.GetOptionalCustomModifiers%2A> and <xref:System.Type.GetRequiredCustomModifiers%2A> methods are provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace. Also see the metadata specification in Partition II of the [Common Language Infrastructure (CLI) documentation](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/).
```

File: xml/System/Type.xml
Comment (carlossanlop) [inline xml/System/Type.xml:8955]:
```suggestion
 The <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetRequiredCustomModifiers%2A> methods are provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace. Also see the metadata specification in Partition II of the [Common Language Infrastructure (CLI) documentation](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/).
```

**Review by carlossanlop — APPROVED** [review]
Need to fix the CI warnings. Once it's green, we can merge.

**Comment by carlossanlop** [comment]
BTW this repo requires resolving comments so that we can unblock merging.

**Pull request merged**
